strawberryfields/cli/__init__.py

```diff
@@ -0,0 +1,219 @@
+    command line interface.
+
+    Returns:
+        ArgumentParser: an argument parser object that defines the related
+            options
+    """
+    parser = argparse.ArgumentParser(
+        usage="starship <command> [<args>]",
+        description="These are common options when working on the Xanadu cloud platform.",
+    )
+
+    # Setting a title for the general options (requires setting a private
+    # attribute)
+    parser._optionals.title = "General Options"
+
+    # Adding the pinging general option
+    parser.add_argument(
+        "--ping", "-p", action="store_true", help="Tests the connection to the remote backend."
+    )
+
+    # Adding subparsers configure and input
+    subparsers = parser.add_subparsers(title="Commands")
+
+    # Adding the configure subparser
+    configure_parser = subparsers.add_parser(
+        "configure", help="Configure each detail of the API connection."
+    )
+    configure_parser.set_defaults(func=configure)
+
+    configure_parser.add_argument(
+        "--token",
+        "-t",
+        type=str,
+        help="Configure only the token by using defaults for other options.",
+    )
+    configure_parser.add_argument(
+        "--local", "-l", action="store_true", help="Create the configure for the project."
+    )
+
+    # Adding the input subparser
+    run_parser = subparsers.add_parser("run", help="Run a blackbird script.")
+    run_parser.add_argument(
+        "input", type=str, help="The input blackbird script to run.",
+    )
+    run_parser.set_defaults(func=run_blackbird_script)
+    run_parser.add_argument(
+        "--output",
+        "-o",
+        help="Path to the output file, where the results of the program will be written (stdout by default).",
+    )
+
+    return parser
+
+
+def configure(args):
+    """An auxiliary function for configuring the API connection to the Xanadu
+    cloud platform.
+
+    Supports configuration by:
+    * only specifying the token and using default configuration options;
+    * specifying every configuration option one by one.
+
+    Related arguments:
+    * token: the authentication token to use
+    * local: whether or not to create the configuration file locally
+
+    Args:
+        args (ArgumentParser): arguments that were specified on the command
+            line stored as attributes in an argument parser object
+    """
+    if args.token:
+        kwargs = {"authentication_token": args.token}
+    else:
+        kwargs = configure_everything()
+
+    if args.local:
+        store_account(**kwargs, location="local")
+    else:
+        store_account(**kwargs)
+
+
+def ping():
+    """Tests the connection to the remote backend."""
+    if Connection().ping():
+        sys.stdout.write("You have successfully authenticated to the platform!\n")
+    else:
+        sys.stdout.write("There was a problem when authenticating to the platform!\n")
+    sys.exit()
+
+
+def configure_everything():
```

Comment:
Nitpick: the name `configuration_wizard` might be more descriptive; `configure_everything` somewhat hides the interactivity of this function, and the fact that the main logic in this function is the wizard rather than determining defaults.

Reply:
Oh that's a great point, I'll update it! :blush: